### DISCUSSION TITLE
JSON data type in Azure sql server

### DISCUSSION THREAD

**Discussion by mayank-adyne** [sqlalchemy/sqlalchemy#12128] [Usage Questions] [answered]
Now Azure SQL server natively supports JSON data type. While in SQLAlchemy JSON type is mapped to nvarchar. Native support has many benefits over nvarchar. How can I use the native json type with SQLAlchemy? 

Link on the new data type.
https://learn.microsoft.com/en-us/sql/t-sql/data-types/json-data-type?view=azuresqldb-current&viewFallbackFrom=sql-server-ver16

**Comment by zzzeek** [comment, marked as answer]
use a [compile rule](https://docs.sqlalchemy.org/en/20/core/compiler.html#changing-compilation-of-types)

```py
from sqlalchemy.types import JSON
from sqlalchemy.ext.compiler import compiles



@compiles(JSON, 'mssql')
def compile_json(element, compiler, **kw):
    return "JSON"

```

**Reply by mayank-adyne** [reply]
Thanks. Will I need to change any handling of the data as well?

**Reply by zzzeek** [reply]
probably not?  Id try it out and run some tests

**Reply by mayank-adyne** [reply]
Thanks @zzzeek this does the trick.
